src/sinks/file/mod.rs

```diff
@@ -174,9 +181,18 @@ impl OutFile {
 impl SinkConfig for FileSinkConfig {
     async fn build(
         &self,
-        _cx: SinkContext,
+        cx: SinkContext,
     ) -> crate::Result<(super::VectorSink, super::Healthcheck)> {
-        let sink = FileSink::new(self)?;
+        let timezone = self.timezone.or(cx.globals.timezone);
+        let config = Self {
+            path: self.path.clone(),
+            idle_timeout: self.idle_timeout,
+            encoding: self.encoding.clone(),
+            compression: self.compression,
+            acknowledgements: self.acknowledgements,
+            timezone,
+        };
+        let sink = FileSink::new(&config)?;
```

Comment:
```suggestion
        let sink = FileSink::new(&config, cx)?;
```

<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏ It feels a bit excessive to recreate the entire config here just to default the timezone. I would prefer it if we could pass the `SinkContext` to `FileSink::new` so that it could access `cx.globals.timezone` there at the the point where it resolves the offset.

Reply:
If we don't do it in the `FileSinkConfig`, there is no way we can override the globals `timezone` as we don't have access to the config inside `FileSink`.

Reply:
You can pass `SinkContext` into `FileSink::new` and then

```rust
    pub fn new(config: &FileSinkConfig, cx: SinkContext) -> crate::Result<Self> {
        let transformer = config.encoding.transformer();
        let (framer, serializer) = config.encoding.build(SinkType::StreamBased)?;
        let encoder = Encoder::<Framer>::new(framer, serializer);

        let offset = config.timezone.or(cx.globals.timezone).and_then(timezone_to_offset);
```

Reply:
Got it. Thanks